src/siteSettings/views/index.tsx

```diff
@@ -35,167 +32,149 @@ export const SiteSettings: React.FC<SiteSettingsProps> = ({ params }) => {
-                      ...(updateShopSettingsOpts.data?.shopDomainUpdate
-                        .errors || []),
-                      ...(updateShopSettingsOpts.data?.shopSettingsUpdate
-                        .errors || []),
-                      ...(updateShopSettingsOpts.data?.shopAddressUpdate
-                        .errors || [])
-                    ];
-                    const loading =
-                      siteSettings.loading ||
-                      addAuthorizationKeyOpts.loading ||
-                      updateShopSettingsOpts.loading;
+  const handleAuthorizationKeyDelete = async (
+    keyType: AuthorizationKeyType
+  ) => {
+    authorizationKeyDelete({
+      variables: { keyType }
+    });
+  };
 
-                    const handleAuthenticationKeyAdd = (
-                      data: SiteSettingsKeyDialogForm
-                    ) =>
-                      addAuthorizationKey({
-                        variables: {
-                          input: {
-                            key: data.key,
-                            password: data.password
-                          },
-                          keyType: data.type
-                        }
-                      });
-                    const handleUpdateShopSettings = async (
-                      data: SiteSettingsPageFormData
-                    ) => {
-                      const addressInput = areAddressInputFieldsModified(data)
-                        ? {
-                            city: data.city,
-                            companyName: data.companyName,
-                            country: findInEnum(data.country, CountryCode),
-                            countryArea: data.countryArea,
-                            phone: data.phone,
-                            postalCode: data.postalCode,
-                            streetAddress1: data.streetAddress1,
-                            streetAddress2: data.streetAddress2
-                          }
-                        : {
-                            companyName: data.companyName
-                          };
-                      const result = await updateShopSettings({
-                        variables: {
-                          addressInput,
-                          shopDomainInput: {
-                            domain: data.domain,
-                            name: data.name
-                          },
-                          shopSettingsInput: {
-                            customerSetPasswordUrl: data.customerSetPasswordUrl,
-                            defaultMailSenderAddress:
-                              data.defaultMailSenderAddress,
-                            defaultMailSenderName: data.defaultMailSenderName,
-                            description: data.description
-                          }
-                        }
-                      });
+  const handleShopSettingsUpdate = async (data: SiteSettingsPageFormData) => {
```

Comment:
Previously this handler had returned errors and consequently had `SubmitPromise` as return type. `SubmitPromise` would be used then by `useForm` hook to indicate change state. Why now errors are not returned?

Reply:
Errors are passed separately in line 137. You can test changes yourself - errors are displayed as intended.

Reply:
It allows form to reset it's dirty status, disabling `save` button after successful update

Reply:
Good to know, changing.